worker/buildbot_worker/util/process.py

```diff
@@ -17,46 +17,45 @@
 
 import os
 import re
+from typing import Mapping
```

Comment:
This commit should have been multiple. It's hard to understand what in compute_environ is being changed. If the refactor was a series of smaller changes then it would be trivial to understand what's happening.

Reply:
@tdesveaux I actually started splitting the commit into multiple as part of review... And yeah, it indeed is not reviewable in current form.

Reply:
oh sorry, i stepped on your toes. Anyway, I thought that I would write a test just to check if my hunch that hard to review code may contain bugs. And indeed, some of the newly written tests fail.

I split the commit only a little though.

I pushed to your branch so that you get the new tests.

Reply:
Thanks for the tests!

Did not see test failure though.
I think it was due to a miss in the changes you made to make the function testable or am I missing ? ([here](https://github.com/buildbot/buildbot/compare/27a1cfb3d3aa673625538b1c4f9bd3777e99a578..19fb3df7180cb33bb2cfeb07bed59a20a00f8159#diff-2ef760dc3aea31a812c71cb90661e548170195d9059eb401ca96c2c928d01a9fR54)

In any case, you were completely right to ask for the commit to be split.
Always feel free to ask for anything that make your reviews easier, I welcome it. (and it make history better to blame too in this case).

Reply:
PR might be a bit large now.
Let me know if you'd prefer I move the `compute_environ` stuff to another.